app/Services/EmployeeService.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services;
+
+use App\Helpers\PhotoHelper;
+use App\Http\Requests\IndexRequest;
+use App\Models\User;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Support\Facades\Hash;
+use Spatie\Permission\Models\Role;
+
+class EmployeeService
+{
+    public function __construct(protected SearchService $searchService)
+    {
+    }
+
+    public function getAll(IndexRequest $indexRequest): object
+    {
+        return $this->searchService->searchItems(new User(), $indexRequest);
+    }
+
+    public function update(User $user, FormRequest $request): void
+    {
+        $validatedData = $request->validated();
+
+        if ($request->hasFile('avatar') && $request->file('avatar')->isValid()) {
+            if ($user->avatar)
+                PhotoHelper::deletePreviousPhoto($user->photo);
+            $validatedData['avatar'] = $request->file('avatar')->store('images/avatars', 'public');
+        }
+
+        $user->update($validatedData);
+    }
+
+    public function changePassword(User $user, FormRequest $request): void
+    {
+        $user->update(['password' => Hash::make($request->validated())]);
+    }
+
+    public function checkAndSetBlock(User $user): void
```

Comment:
zmieniłbym nazwę na toggleBlock, wydaje mi się że checkAndSetBlock, sugeruje że coś jest sprawdzane a potem zawsze jest blokowany, a tak to wiadomo, że jest odwaracanie obecnego statusu